### PR TITLE
Adds a Grammar.md file, to contain markup language specification.

### LINKED ISSUE #3: Add Markup Language Specification

### PR DESCRIPTION
**Pull request by tylercecil**
Closes #3. The language specification I've written doesn't include any of the more advanced potential features, such as movement lines, theta roles, or possible meta-data marking. I might want to do everything at once, rather than change the parser around a bunch as we reconfigure things, but because it's so simple I'm comfortable incrementally changing the syntax. I'm not sure when we'll settle on a stable syntax, but it shouldn't take *that* long to get there.

### CODE REVIEW

**Comment by tylercecil** [comment]
I've actually put a lot of thought into changes for this. I'll commit in an hour or so.

File: Grammar.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+```
+Node     := "[" "]" | "[" Node_T {"." Node_T} ( NodeList | NodeData ) "]".
+Node_T   := word [(sub [sup]) | (sup [sub])].
+NodeList := { Node }.
+NodeData := ( ["*"] word {word | DataSymbols} | Null ).
+sub      := "_" word.
+sup      := "^" word.
+Null     := "/".
+DataSymbols := "." | "*" | "/" | "_" | "^".
```
Comment (tylercecil) [inline Grammar.md:89]: Note, this means nodes can't contain `[` or `]`. I don't know if I care... but I can add escape sequences for them, I guess?

**Comment by tylercecil** [comment]
The language, as written, is linear (with the exception of empty nodes, but that wont change the implementation at all), which I think is important. I have essentially 3 major features I haven't addressed, though I have an idea of how they should look.

1. Movement lines. I'm considering `[Z [X this gets moved]a> [Y to here]a< ]`
2. Assigning classes to nodes, to allow for custom CSS. Maybe `[X is a node of class].class`
3. Writing extra lines of data (like theta roles / glosses / classes). (There's lots of ways)

2 and 3 could possibly be combined somehow. Maybe all 3. I'm fine pushing these features off, I think.

File: Grammar.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+For chains of non branching nodes, a `.` may be placed between node types. This
+is equivelant to `[NP [D the] [AdjP [Adj big]] [N dog]]`.
+
+### Collapsed Tree
+
+```
+[NP* the big red dog]
+```
+
+To mark an unknown subtree, an `*` is used.
```
Comment (duncgibbs) [inline Grammar.md:37]: 👍

File: Grammar.md (hunk 1 above)
Comment (duncgibbs) [reply Grammar.md:89]: I think that's fine. Escape can be added later if anyone ever needs to use square brackets in a sentence. Which seems unlikely.

**Pull request merged**